tools/cldr-code/src/main/java/org/unicode/cldr/test/ExampleGenerator.java

```diff
@@ -747,18 +750,22 @@ private String handleFormatUnit(XPathParts parts, String value) {
          */
         @SuppressWarnings("deprecation")
         FixedDecimal amount = getBest(Count.valueOf(count));
-        DecimalFormat numberFormat = null;
-        if (amount != null) {
-            numberFormat = icuServiceBuilder.getNumberFormat(1);
-            examples.add(format(value, backgroundStartSymbol + numberFormat.format(amount) + backgroundEndSymbol));
+        if (amount == null) {
+            return null;
         }
+        DecimalFormat numberFormat = null;
+        String formattedAmount = null;
+        numberFormat = icuServiceBuilder.getNumberFormat(1);
+        formattedAmount = numberFormat.format(amount);
```

Comment:
does the setting these to null right before setting them to something else serve a purpose?